client/packages/host/src/components/Android.tsx

```diff
@@ -122,15 +135,16 @@ export const Android = () => {
     if (mode === NativeMode.Server) {
       const localServer = servers.find(server => server.isLocal);
       if (localServer) {
-        connectToServer(localServer);
+        const path = !token ? 'login' : '';
+        connectToServer({ ...localServer, path });
       }
     }
   }, [mode, servers]);
 
   useEffect(() => {
     if (
       mode === NativeMode.Client &&
-      (!previousServer || connectToPreviousTimedOut)
+      (!previousServer?.ip || connectToPreviousTimedOut)
```

Comment:
slightly clumsy - this is because there's no removePreference, and I'm setting the previousServer to an empty object when clearing it